docker/ubuntu14.04/Dockerfile

```diff
@@ -0,0 +1,42 @@
+FROM ubuntu:14.04
```

Comment:
Thanks for adding these `Dockerfile`. This would help us to test changes locally before raising PR. Just thinking, if we can use these Dockerfiles in the CI environment - i.e.. running our builds within these container instead of directly on github virtual machines.  Github action does support it : https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/creating-a-docker-container-action.

Reply:
Yes, possibly. I was thinking about this too. For now my focus was on improving / speeding up the dev experience. I did this for Windows-Docker, but I will add corresponding script for Mac and Linux docker as well. Right now the set of files allows us to capture wide spectrum of execution environments:
- Ubuntu 14.04 - gcc-4.8
- Ubuntu 14.04 - gcc-5
- Ubuntu 18.04 - gcc-7
- Ubuntu 20.04 - gcc-8

Incremental builds, the way how the docker images are now set up  - with pre-installed latest CMake, are blazing fast! The goal is to build, including ability to incrementally build, and run all the tests in one run.

Reply:
I logged it in #393